rust_src/src/buffers.rs

```diff
@@ -320,6 +335,57 @@ pub fn overlay_buffer(overlay: LispObject) -> LispObject {
     marker_buffer(marker)
 }
 
+pub fn recenter_overlay_lists(mut buf: LispBufferRef, pos: ptrdiff_t) {
+    // let mut overlay: LispObject;
+    let (mut prev, mut tail, mut next): (
+        Option<LispOverlayRef>,
+        LispOverlayRef,
+        Option<LispOverlayRef>,
+    ) = (
+        None,
+        LispOverlayRef::new((*buf).overlays_before as *mut Lisp_Overlay),
```

Comment:
Does this allocate? Or just wraps a pointer with a better API?

Reply:
A `LispOverlayRef` is a Rust type that wraps a C pointer. So yes, an allocation is made but not for the C data. We treat the `ExternalPtr` wrappers as free in the Rust code.

Reply:
`new` in Rust is an idiom for an initializer. It does not mean a heap allocation has necessarily been made.